From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Thu, 28 Oct 2021 23:05:57 +0300
Subject: [PATCH] (halium) minimize com.android.art apex build

Include only libs other native components depend on, such as icu
(for libmedia.so) and libexpat.

Change-Id: I34850535fe13a07354fdbd82d0f2bbbbc0336f6f
---
 build/apex/Android.bp | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/build/apex/Android.bp b/build/apex/Android.bp
index c5acc915d4..38193bac88 100644
--- a/build/apex/Android.bp
+++ b/build/apex/Android.bp
@@ -238,7 +238,6 @@ apex_defaults {
     defaults: ["com.android.art-defaults"],
     compile_multilib: "both",
     manifest: "manifest-art.json",
-    bootclasspath_fragments: ["art-bootclasspath-fragment"],
     compat_configs: ["libcore-platform-compat-config"],
     java_libs: libcore_java_libs,
     native_shared_libs: art_runtime_base_native_shared_libs +
@@ -254,7 +253,16 @@ apex_defaults {
                 art_tools_common_binaries +
                 art_tools_device_only_binaries,
         },
-    },
+    },*/
+    // Halium: build only libs other native components depend on
+    native_shared_libs: [
+        "libandroidicu",
+        "libandroidio",
+        "libexpat",
+        "libicui18n",
+        "libicuuc",
+        "libnativehelper",
+    ],
     key: "com.android.art.key",
     required: [
         "com.android.i18n",
@@ -295,7 +303,7 @@ apex_defaults {
                 art_tools_debug_binaries +
                 art_tools_debug_device_only_binaries,
         },
-    },
+    },*/
 }
 
 apex_test {
-- 
2.25.1

